internal/manifests/config.go

```diff
@@ -11,34 +10,32 @@ import (
 )
 
 // LokiConfigMap creates the single configmap containing the loki configuration for the whole cluster
-func LokiConfigMap(opt Options) (*corev1.ConfigMap, string, error) {
+func LokiConfigMap(opt Options) (*corev1.ConfigMap, config.CompareResult, error) {
 	cfg := ConfigOptions(opt)
 	c, rc, err := config.Build(cfg)
 	if err != nil {
-		return nil, "", err
+		return nil, nil, err
 	}
 
-	s := sha1.New()
-	_, err = s.Write(c)
+	res, err := config.Compare(opt.Config.Config, c)
```

Comment:
do we actually need to know what the current config is? wouldn't just knowing the resulting hashes for what we generate be enough?